Tests/TuistGeneratorTests/Generator/LinkGeneratorTests.swift

```diff
@@ -88,6 +88,46 @@ final class LinkGeneratorErrorTests: XCTestCase {
             XCTAssertEqual($0 as? LinkGeneratorError, LinkGeneratorError.missingProduct(name: "Test"))
         }
     }
+    
+    func test_generateEmbedPhase_setupEmbedFrameworksBuildPhase_whenXCFrameworkIsPresent() throws {
+        // Given
+        var dependencies: [DependencyReference] = []
+        dependencies.append(DependencyReference.absolute(AbsolutePath("/Frameworks/Test.xcframework")))
+        let pbxproj = PBXProj()
+        let pbxTarget = PBXNativeTarget(name: "Test")
+        let sourceRootPath = AbsolutePath("/")
+
+        let group = PBXGroup()
+        pbxproj.add(object: group)
+    
+        let from = AbsolutePath("/Frameworks/")
+        let fileAbsolutePath = AbsolutePath("/Frameworks/Test.xcframework")
+        let fileRelativePath = RelativePath("./Test.xcframework")
+        let fileElements = ProjectFileElements()
+        fileElements.addFileElement(from: from,
+                                          fileAbsolutePath: fileAbsolutePath,
+                                          fileRelativePath: fileRelativePath,
+                                          name: nil,
+                                          toGroup: group,
+                                          pbxproj: pbxproj)
+
+        let wakaFile = fileElements.file(path: fileAbsolutePath)
```

Comment:
I wonder if we can benefit from a test helper here to make the test more concise.

```swift
// Given
let xcframeworkPath = AbsolutePath("/Frameworks/Test.xcframework")
let dependencies: [DependencyReference] = [
  .absolute(xcframeworkPath)
]
let fileElements = createFileElements(for: xcframeworkPath)
// ...

```

Note: "waka" was the name of a framework used in another test, there is no need to use it here as the framework we have here is called "test", the name doesn't really impact the test.

Reply:
Agreed, we could make code concise by using a helper.